File: benchmark/output/lua-love/openai/multi_wave_spawner.lua
local playerX = 400
local playerY = 300
local playerSpeed = 250
local playerRadius = 12

local wave = 1
local spawn_timer = 2.0
local enemies_per_wave = 3

local ex, ey, evx, evy, elife = {}, {}, {}, {}, {}
local enemySize = 16
local enemySpeedBase = 80

local game_over = false

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Wave-based Enemy Spawner")
end

function love.update(dt)
    if game_over then return end

    -- Player movement
    if love.keyboard.isDown("left") then
        playerX = playerX - playerSpeed * dt
    end
    if love.keyboard.isDown("right") then
        playerX = playerX + playerSpeed * dt
    end
    if love.keyboard.isDown("up") then
        playerY = playerY - playerSpeed * dt
    end
    if love.keyboard.isDown("down") then
        playerY = playerY + playerSpeed * dt
    end

    -- Spawn enemies
    spawn_timer = spawn_timer - dt
    if spawn_timer <= 0 then
        for i = 1, enemies_per_wave do
            local edge = love.math.random() * 4
            local x, y
            if edge < 1 then
                x = love.math.random() * 800
                y = 0
            elseif edge < 2 then
                x = 800
                y = love.math.random() * 600
            elseif edge < 3 then
                x = love.math.random() * 800
                y = 600
            else
                x = 0
                y = love.math.random() * 600
            end

            local dx = playerX - x
            local dy = playerY - y
            local dist = math.sqrt(dx * dx + dy * dy)
            local speed = enemySpeedBase + wave * 20

            table.insert(ex, x)
            table.insert(ey, y)
            table.insert(evx, (dx / dist) * speed)
            table.insert(evy, (dy / dist) * speed)
            table.insert(elife, 1.0)
        end

        wave = wave + 1
        enemies_per_wave = 3 + wave
        spawn_timer = math.max(0.5, 3.0 - wave * 0.2)
    end

    -- Update enemies
    for i = #ex, 1, -1 do
        if elife[i] > 0 then
            ex[i] = ex[i] + evx[i] * dt
            ey[i] = ey[i] + evy[i] * dt
            elife[i] = elife[i] - dt

            -- Check collision with player
            local dx = playerX - ex[i]
            local dy = playerY - ey[i]
            local distance = math.sqrt(dx * dx + dy * dy)
            if distance < playerRadius + enemySize / 2 then
                game_over = true
            end
        end
    end
end

function love.draw()
    love.graphics.setColor(1, 1, 1)
    love.graphics.circle("fill", playerX, playerY, playerRadius)

    for i = 1, #ex do
        if elife[i] > 0 then
            love.graphics.rectangle("fill", ex[i] - enemySize / 2, ey[i] - enemySize / 2, enemySize, enemySize)
        end
    end

    love.graphics.print("Wave: " .. wave, 10, 10)

    if game_over then
        love.graphics.printf("GAME OVER", 0, love.graphics.getHeight() / 2 - 10, love.graphics.getWidth(), "center")
    end
end

File: benchmark/output/python-pygame/openai/multi_wave_spawner.py
import pygame
import sys
import random
import math

def rand_float(min_val, max_val):
    return random.uniform(min_val, max_val)

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Vibe Game")
clock = pygame.time.Clock()

# Player variables
px, py = 400.0, 300.0
player_speed = 250
player_radius = 12

# Game state variables
wave = 1
spawn_timer = 2.0
enemies_per_wave = 3
game_over = False

# Enemy lists
ex, ey = [], []
evx, evy = [], []
elife = []

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    if not game_over:
        keys = pygame.key.get_pressed()
        if keys[pygame.K_LEFT]:
            px -= player_speed * dt
        if keys[pygame.K_RIGHT]:
            px += player_speed * dt
        if keys[pygame.K_UP]:
            py -= player_speed * dt
        if keys[pygame.K_DOWN]:
            py += player_speed * dt

        # Clamp player position to screen
        px = max(player_radius, min(800 - player_radius, px))
        py = max(player_radius, min(600 - player_radius, py))

        # Spawn enemies
        spawn_timer -= dt
        if spawn_timer <= 0.0:
            for _ in range(enemies_per_wave):
                edge = rand_float(0.0, 4.0)
                if edge < 1.0:
                    ex.append(rand_float(0, 800))
                    ey.append(0)
                elif edge < 2.0:
                    ex.append(800)
                    ey.append(rand_float(0, 600))
                elif edge < 3.0:
                    ex.append(rand_float(0, 800))
                    ey.append(600)
                else:
                    ex.append(0)
                    ey.append(rand_float(0, 600))

                # Calculate direction towards player
                dx = px - ex[-1]
                dy = py - ey[-1]
                dist = math.sqrt(dx**2 + dy**2)
                speed = 80 + wave * 20
                evx.append((dx / dist) * speed)
                evy.append((dy / dist) * speed)
                elife.append(1.0)

            wave += 1
            enemies_per_wave = 3 + wave
            spawn_timer = max(0.5, 3.0 - wave * 0.2)

        # Update enemies
        for i in range(len(ex)):
            if elife[i] > 0:
                ex[i] += evx[i] * dt
                ey[i] += evy[i] * dt
                elife[i] -= dt

                # Check collision with player
                if math.sqrt((px - ex[i])**2 + (py - ey[i])**2) < player_radius + 8:
                    game_over = True

    # Draw everything
    screen.fill((0, 0, 0))
    pygame.draw.circle(screen, (255, 255, 255), (int(px), int(py)), player_radius)

    for i in range(len(ex)):
        if elife[i] > 0:
            pygame.draw.rect(screen, (255, 255, 255), (int(ex[i]) - 8, int(ey[i]) - 8, 16, 16))

    font = pygame.font.Font(None, 36)
    wave_text = font.render(f"Wave: {wave}", True, (255, 255, 255))
    screen.blit(wave_text, (10, 10))

    if game_over:
        game_over_text = font.render("GAME OVER", True, (255, 0, 0))
        screen.blit(game_over_text, (400 - game_over_text.get_width() // 2, 300 - game_over_text.get_height() // 2))

    pygame.display.flip()

pygame.quit()
sys.exit()